webapp-django/crashstats/crashstats/tests/test_views.py

```diff
@@ -407,42 +407,74 @@ def mocked_get(url, params, **options):
                  }
                       """ % {'end_date': now.strftime('%Y-%m-%d'),
                              'yesterday': yesterday.strftime('%Y-%m-%d')})
-            if '/supersearch/fields/' in url:
-                results = copy.copy(SUPERSEARCH_FIELDS_MOCKED_RESULTS)
-                # to be realistic we want to introduce some dupes
-                # that have a different key but its `in_database_name`
-                # is one that is already in the hardcoded list (the
-                # baseline)
-                assert 'accessibility' not in results
-                results['accessibility'] = {
-                    'name': 'accessibility',
-                    'query_type': 'string',
-                    'namespace': 'raw_crash',
-                    'form_field_choices': None,
-                    'permissions_needed': [],
-                    'default_value': None,
-                    'is_exposed': True,
-                    'is_returned': True,
-                    'is_mandatory': False,
-                    'in_database_name': 'Accessibility',
-                }
-                return Response(results)
             raise NotImplementedError(url)
 
         rget.side_effect = mocked_get
 
+        def mocked_supersearchfields(**params):
+            results = copy.copy(SUPERSEARCH_FIELDS_MOCKED_RESULTS)
+            # to be realistic we want to introduce some dupes
+            # that have a different key but its `in_database_name`
+            # is one that is already in the hardcoded list (the
+            # baseline)
+            assert 'accessibility' not in results
+            results['accessibility'] = {
+                'name': 'accessibility',
+                'query_type': 'string',
+                'namespace': 'raw_crash',
+                'form_field_choices': None,
+                'permissions_needed': [],
+                'default_value': None,
+                'is_exposed': True,
+                'is_returned': True,
+                'is_mandatory': False,
+                'in_database_name': 'Accessibility',
+            }
+            return results
+
+        self.supersearch = mock.MagicMock()
+        self.supersearchfields = mock.MagicMock()
+        self.missingfields = mock.MagicMock()
+
+        from crashstats.supersearch.models import (
+            SuperSearch,
+            SuperSearchFields,
+            SuperSearchField,
+            SuperSearchMissingFields,
+        )
+        # monkey-patch a class attribute
+        SuperSearchFields.implementation = self.supersearchfields
+        SuperSearchField.implementation = self.supersearchfields
+        SuperSearchMissingFields.implementation = self.missingfields
+        SuperSearch.implementation = self.supersearch
```

Comment:
Are we going to do this for every class that we move to django? If so it doesn't seem like a good path to me, it will quickly become a huge mess. Plus that means every time you test something you have to navigate to this base class to make some changes. I agree it's a good thing to have that for the SuperSearchFields.get function, which is being used in a lot of different places, and maybe it also makes sense for SuperSearch since we will use it more and more, but I don't think SuperSearchMissingFields should be there. That one should be in its own, encapsulated test.